### PR TITLE
handling missing update stanza for group

### PR DESCRIPTION
**Pull request by burdandrei**

### CODE REVIEW

File: client/job_scaling_policies.go [hunk 1]
```diff
@@ -115,6 +115,11 @@ func (c *nomadClient) jobScalingPolicyProcessor(jobID string, scaling *structs.J
 		// If all keys were matched we update the job scaling policy struct with the
 		// information.
 		if len(missedKeys) == 0 {
+			if group.Update == nil {
```
Comment (ericwestfall) [inline client/job_scaling_policies.go]: I believe we should check at the job level as well as the group level for an `update` stanza. We would need to determine if the job or any group within the job has an `update` stanza before declaring the job not fit for scaling with verification.

Comment (burdandrei) [reply client/job_scaling_policies.go]: the thing is it if you got update stanza on job level, group inherits it, so no need to check it twice

Comment (ericwestfall) [reply client/job_scaling_policies.go]: Ok sounds good, I agree with all that then. 

Can we move this check up to before line `95`, there is no need to check the meta keys if we decide the group is not scalable. Just clean up the error message to remove the bits about the meta keys and maybe make a reference that Replicator currently requires all scalable jobs to include an `update` stanza.

Comment (burdandrei) [reply client/job_scaling_policies.go]: done

**Comment by ericwestfall** [comment]
Thanks so much for the bug report and PR @burdandrei!!

**Pull request merged**